django/utils/url_config.py

```diff
@@ -0,0 +1,65 @@
+import inspect
+from collections import defaultdict
+
+from django.core.exceptions import ImproperlyConfigured
+from django.utils import module_loading
+
+BACKENDS = defaultdict(dict)
+
+
+def add_backend(backend_type, scheme, path):
+    if scheme in BACKENDS[backend_type]:
+        raise ImproperlyConfigured('Scheme {0} has already been registered as a {1} handler'.format(scheme,
+                                                                                                    backend_type))
+    BACKENDS[backend_type][scheme] = path
+
+
+def get_backend(backend_type, scheme):
+    if scheme not in BACKENDS[backend_type]:
+        raise ImproperlyConfigured('No {0} handler configured for {1}'.format(backend_type, scheme))
+    return BACKENDS[backend_type][scheme]
+
+
+def configure(backend_type, value):
+    scheme = value.split(':', 1)[0]
+
+    backend_path = get_backend(backend_type, scheme)
+    handler = module_loading.import_string(backend_path + '.base.DatabaseWrapper')
+
+    if not hasattr(handler, 'config_from_url'):
+        raise RuntimeError('{0} has no config_from_url method'.format(backend_path))
+
+    if not inspect.ismethod(handler.config_from_url):
```

Comment:
Would `isinstance(handler.config_from_url, classmethod)` work? If so it has the advantage of not requiring `inspect`.